src/main/java/ch/njol/skript/effects/EffOpenInventory.java

```diff
@@ -34,125 +36,201 @@
-				"close [the] inventory [view] (to|of|for) %players%", "close %players%'[s] inventory [view]");
+				"show %inventory/inventorytype% (to|for) %players%",
+				"open [a[n]] " + OpenableInventorySyntax.construct() + "[view|window|inventory] (to|for) %players%",
+				"close [the] inventory [view] (of|for) %players%",
+				"close %players%'[s] inventory [view]");
 	}
-	
+
+	private boolean open;
+
 	@Nullable
-	private Expression<?> invi;
-	
-	boolean open;
-	private int invType;
-	
-	@SuppressWarnings("null")
+	private OpenableInventorySyntax syntax;
+
+	@Nullable
+	private Expression<?> inventory;
 	private Expression<Player> players;
-	
-	@SuppressWarnings({"unchecked", "null"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		int openFlag = 0;
-		if(parseResult.mark >= 5) {
-			openFlag = parseResult.mark ^ 5;
-			invType = DISPENSER;
-		} else if(parseResult.mark >= 4) {
-			openFlag = parseResult.mark ^ 4;
-			invType = DROPPER;
-		} else if(parseResult.mark >= 3) {
-			openFlag = parseResult.mark ^ 3;
-			invType = HOPPER;
-		} else if (parseResult.mark >= 2) {
-			openFlag = parseResult.mark ^ 2;
-			invType = ANVIL;
-		} else if (parseResult.mark >= 1) {
-			openFlag = parseResult.mark ^ 1;
-			invType = CHEST;
-		} else if (parseResult.mark >= 0) {
-			invType = WORKBENCH;
-			openFlag = parseResult.mark ^ 0;
-		} else {
-			openFlag = parseResult.mark;
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (matchedPattern == 1) {
+			open = true;
+			assert !parseResult.tags.isEmpty() : "Syntax incorrectly handled parse result tags.";
+			syntax = OpenableInventorySyntax.valueOf(parseResult.tags.get(0).toUpperCase(Locale.ENGLISH));
+			if (!Skript.isRunningMinecraft(syntax.getVersion())) {
+				Skript.error("Opening an inventory of type '" + syntax.toString().toLowerCase(Locale.ENGLISH) + "' is only present on Minecraft version " + syntax.getVersion());
+				return false;
+			}
+			if (!syntax.doesMethodExist()) {
+				Skript.error(syntax.getMethodError());
+				return false;
+			}
 		}
-		
-		open = matchedPattern == 0;
-		invi = open ? exprs[0] : null;
+		inventory = matchedPattern == 0 ? exprs[0] : null;
 		players = (Expression<Player>) exprs[exprs.length - 1];
-		if (openFlag == 1 && invi != null) {
-			Skript.warning("Using 'show' inventory instead of 'open' is not recommended as it will eventually show an unmodifiable view of the inventory in the future.");
+		if (inventory instanceof Literal) {
+			Literal<?> literal = (Literal<?>) inventory;
+			Object object = literal.getSingle();
+			if (object instanceof InventoryType && !((InventoryType) object).isCreatable()) {
+				Skript.error("You can't open a '" + literal.toString() + "' inventory to players. It's not creatable.");
```

Comment:
What does the output of literal.toString() actually look like?

Reply:
The inventory type? `chest` `dropper` etc. It's the same as doing toString(event, debug) but literals allow for no event because they're a literal.